### ISSUE TITLE
Azure Isolated Functions Docker Image Creation Error

### ISSUE THREAD

**Issue by david200111** [Azure/azure-functions-docker#885]
HI,

When I try and create a docker image for an isolated Azure Timer Function using "**docker build . -f DockerFile**", via the command line, in VS2022 I am getting the following error

/usr/share/dotnet/sdk/6.0.407/NuGet.targets(132,5): error MSB4181: The "RestoreTask" task returned false but did not log an error. [/tmp/21213513dffd/WorkerExtension.csproj]

The docker file is

#See https://aka.ms/customizecontainer to learn how to customize your debug container and how Visual Studio uses this Dockerfile to build your images for faster debugging.

FROM mcr.microsoft.com/azure-functions/dotnet-isolated:4-dotnet-isolated6.0 AS base
WORKDIR /app

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["ResearchLibrary.Functions.Functions/nuget.config", "ResearchLibrary.Functions.Functions/"]
COPY ["ResearchLibrary.Functions.Functions/ResearchLibrary.Functions.Functions.csproj", "ResearchLibrary.Functions.Functions/"]
COPY ["ResearchLibrary.Functions.Data/ResearchLibrary.Functions.Data.csproj", "ResearchLibrary.Functions.Data/"]
COPY ["ResearchLibrary.Functions.Helper/ResearchLibrary.Functions.Helper.csproj", "ResearchLibrary.Functions.Helper/"]
COPY ["ResearchLibrary.Functions.Domain/ResearchLibrary.Functions.Domain.csproj", "ResearchLibrary.Functions.Domain/"]
RUN dotnet restore --configfile:ResearchLibrary.Functions.Functions/nuget.config "ResearchLibrary.Functions.Functions/ResearchLibrary.Functions.Functions.csproj"
COPY . .
WORKDIR "/src/ResearchLibrary.Functions.Functions"
RUN dotnet build --configfile:nuget.config "ResearchLibrary.Functions.Functions.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "ResearchLibrary.Functions.Functions.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM base AS final  
WORKDIR /app  
COPY --from=publish /app/publish .  
ENTRYPOINT ["dotnet", "ResearchLibrary.Functions.Functions.dll"]  

Any ideas what I am missing or is causing this

**Comment by vijaykumar911** [comment]
Hi @david200111 will   check   the issue and update you.

**Comment by david200111** [comment]
> Hi @david200111 will check the issue and update you.

Thanks, I think the issue is to do with WorkerExtension.csproj that is built on the fly. This has a dependency on version 4.5.0 of System.Text.Encodings.Web, that is blocked by my company's Nuget feed, as it has a critical vulnerability. Do you know how to get this updated so that it uses a later non-critical version of System.Text.Encodings.Web

**Comment by vijaykumar911** [comment]
you can update the version of System.Text.Encodings.Web used by the WorkerExtension project by modifying the project file (WorkerExtension.csproj).

 open the project file in a text editor . Then, locate the section of the file that lists the project's dependencies. Look for a line that references the System.
To update the version of the package, change the Version attribute to a later, non-critical version.

**Comment by david200111** [comment]
> you can update the version of System.Text.Encodings.Web used by the WorkerExtension project by modifying the project file (WorkerExtension.csproj).
> 
> open the project file in a text editor . Then, locate the section of the file that lists the project's dependencies. Look for a line that references the System. To update the version of the package, change the Version attribute to a later, non-critical version.

Hi,

Thanks for the reply. Problem I have is that this project is created on the fly during the build process, so I don't have access to the .csproj file. The Nuget package that creates it is Microsoft.Azure.Functions.Worker.Sdk

**Comment by vivekjilla** [comment]
Thanks for the report. As identified in the thread, the failure came from the auto-generated WorkerExtension project pulling a package version blocked by your corporate NuGet feed — a local feed/SDK configuration issue rather than an image defect. Closing as resolved; if you still hit this on a current supported image, please open a new issue with your NuGet feed configuration.
